### ISSUE TITLE
View messages marked for deletion

### ISSUE THREAD

**Issue by ner00** [the-djmaze/snappymail#470]
Is it possible to implement a functionality to view messages marked for deletion?

This should ideally be configurable in the admin panel for all accounts, but overridden per account in their respective settings screen.

**Comment by the-djmaze** [comment]
I don't think this is supported by IMAP as messages marked for deletion should be purged.
The question is: why aren't they purged/deleted?

**Comment by ner00** [comment]
I'm going off of the experience with other clients, if there is no immediate expunge then the state is "deleted" (marked for deletion). Given that "marked as deleted" is a state that I mostly find when deleting in Outllook, which may not follow standards.

```
 \Deleted
Message is "deleted" for removal by later EXPUNGE
```
https://datatracker.ietf.org/doc/html/rfc3501#section-2.3.2

**Comment by the-djmaze** [comment]
Ah the `\Deleted` flag.
SnappyMail should still show those messages and you should be able to click on them and open the message.
They have the `.msgflag-\\deleted` CSS class.
```less
	&.msgflag-\\deleted {
		opacity: .7;
		.subjectParent {
			text-decoration: line-through;
		}
	}
```

**Comment by ner00** [comment]
Unfortunately they don't appear on the list at all, they do on RainLoop and Roundcube though.

**Comment by the-djmaze** [comment]
Open `snappymail/v/0.0.0/app/libraries/MailSo/Config.php` and set
```php
	public static $MessageListUndeletedOnly = false;
```

This should work, but the behaviour is unpredictable due to how \\Deleted and pruning is setup in your specific IMAP daemon.

**Comment by ner00** [comment]
This worked for SnappyMail. As for RainLoop, it seems to ignore this setting altogether and always behaves as if it were set to `false`.

Any chance we could map this to a setting in the interface? Or is it too unstable/unpredictable to use at all?

Some users have multiple mail clients for their account across devices, some clients will permanently delete a message when "deleted" or moved, others will mark the messages for deletion but not actually expunge them. 

As an example, Roundcube has a setting (Server Settings) - "Flag the message for deletion instead of delete". This is one of the reasons why Roundcube also has the "Compact" option that can be used in any folder, this essentially expunges every message flagged for deletion - that is a different matter, but tied to this context.

**Comment by the-djmaze** [comment]
> Any chance we could map this to a setting in the interface? Or is it too unstable/unpredictable to use at all?

When using, some people question: "Why do i see messages marked for deletion?"
When using, some people question: "Why can't i view messages marked for deletion?"
When not using, some people question: "View messages marked for deletion?"

So, the "unpredictable" behaviour is the "Why can't i view messages marked for deletion?"
And is because of a few things happening:
1. client 1 marks message for deletion
2. client 2 shows that (cached)
3. client 3 "compacr"/expunge the list
4. client 1/2 still think the message exists, but this false due to client 3

So, basically there are always questions raised by people who have no clue how IMAP works, or just say "damn computer" and get coffee and work on something else.

However, we could make an option in the user account settings to have a "(Un)hide messages marked for deletion?""

**Comment by ner00** [comment]
> However, we could make an option in the user account settings to have an "(Un)hide messages marked for deletion?""

This would be nice, as suggested initially. Thanks for considering.

**Comment by the-djmaze** [comment]
Does the solution in v2.17.1 solve your problem?
https://github.com/the-djmaze/snappymail/releases/tag/v2.17.1

**Comment by ner00** [comment]
> Does the solution in v2.17.1 solve your problem? https://github.com/the-djmaze/snappymail/releases/tag/v2.17.1

I would say 'yes' if the setting actually worked. As it stands, it's showing me messages marked for deletion regardless of the actual setting `Hide messages marked for deletion`.

**Comment by ner00** [comment]
Now working as expected.

**Comment by the-djmaze** [comment]
P.S. in the search field you can now type DELETED or UNDELETED to see a list of each.
I forgot to mention that.

**Comment by ner00** [comment]
Question though, what if you actually want to find messages with those words in it? Not trying to be nitpicky here, just wondering.

**Comment by the-djmaze** [comment]
That fails unless you use the advanced search or type `body=unanswered`
